xchange-blockchain/src/main/java/org/knowm/xchange/blockchain/service/BlockchainTradeService.java

```diff
@@ -0,0 +1,177 @@
+package org.knowm.xchange.blockchain.service;
+
+import org.knowm.xchange.blockchain.BlockchainAdapters;
+import org.knowm.xchange.blockchain.BlockchainAuthenticated;
+import org.knowm.xchange.blockchain.BlockchainErrorAdapter;
+import org.knowm.xchange.blockchain.BlockchainExchange;
+import org.knowm.xchange.blockchain.dto.BlockchainException;
+import org.knowm.xchange.blockchain.dto.trade.BlockchainOrder;
+import org.knowm.xchange.blockchain.params.BlockchainTradeHistoryParams;
+import org.knowm.xchange.client.ResilienceRegistries;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.account.OpenPositions;
+import org.knowm.xchange.dto.trade.*;
+import org.knowm.xchange.exceptions.ExchangeException;
+import org.knowm.xchange.exceptions.NotYetImplementedForExchangeException;
+import org.knowm.xchange.service.trade.TradeService;
+import org.knowm.xchange.service.trade.params.*;
+import org.knowm.xchange.service.trade.params.orders.*;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static org.knowm.xchange.blockchain.BlockchainConstants.CURRENCY_PAIR_EXCEPTION;
+import static org.knowm.xchange.blockchain.BlockchainConstants.NOT_IMPLEMENTED_YET;
+
+public class BlockchainTradeService extends BlockchainTradeServiceRaw implements TradeService {
+
+    public BlockchainTradeService(BlockchainExchange exchange, BlockchainAuthenticated blockchainApi, ResilienceRegistries resilienceRegistries) {
+        super(exchange, blockchainApi, resilienceRegistries);
+    }
+
+    @Override
+    public OpenOrders getOpenOrders() throws IOException {
+        return getOpenOrders(createOpenOrdersParams());
+    }
+
+    @Override
+    public OpenOrders getOpenOrders(OpenOrdersParams params) throws IOException {
+        try {
+            return BlockchainAdapters.toOpenOrders(this.getOrders());
+        } catch (BlockchainException e) {
+            throw BlockchainErrorAdapter.adapt(e);
+        }
+    }
+
+    @Override
+    public String placeLimitOrder(LimitOrder limitOrder) throws IOException {
+        Long id = this.postOrder(BlockchainAdapters.toBlockchainLimitOrder(limitOrder)).getExOrdId();
+        if (id != null) return Long.toString(id);
```

Comment:
Because in the case that the order is rejected the id is null and ` return Long.toString(id)`;  it would give error if I do not control the nullability.

Reply:
Take a look at the javadoc. I think if the order is rejected you should thrown an exception instead of returning a null value

Reply:
I can try something like this: 

```
@Override
    public String placeMarketOrder(MarketOrder marketOrder) throws IOException {
        try {
           Long id = this.postOrder(BlockchainAdapters.toBlockchainMarketOrder(marketOrder)).getExOrdId();
           if (id == null) throw new ExchangeException(ORDER_EXCEPTION);
           return Long.toString(id);
        } catch (BlockchainException e) {
            throw BlockchainErrorAdapter.adapt(e);
        }
    }
```

But when I run the tests in BlockchainTradeDemo, in placeMarketOrder the order returns id=null and status REJECTED.
So I get the exception and the rest of the tests no longer run.